WalletWasabi.Gui/MainWindow.xaml.cs

```diff
@@ -14,17 +15,22 @@
 using System.Threading.Tasks;
 using WalletWasabi.Gui.Controls.WalletExplorer;
 using WalletWasabi.Gui.Dialogs;
-using WalletWasabi.Gui.Tabs.WalletManager;
 using WalletWasabi.Gui.ViewModels;
 using WalletWasabi.Logging;
+using WalletWasabi.Services.Terminate;
 
 namespace WalletWasabi.Gui
 {
 	public class MainWindow : MetroWindow
 	{
+		private const int ClosingStateNotClosing = 0;
```

Comment:
We use these constants quite often. It would be great to wrap the mechanism in a class to be generally useful. Something like `ComponentState`.

Reply:
I agree.